src/analysis/FVA/fluxVariability.m

```diff
@@ -370,73 +403,72 @@
 
 function [Flux,V] = calcSolForEntry(model,rxnNameList,i,LPproblem,parallelMode, method, allowLoops, printLevel, minNorm, cpxControl, sol)
 
-    %get Number of reactions
-    nRxns = numel(model.rxns);
-    %Set the correct objective
-    LPproblem.c = double(ismember(model.rxns,rxnNameList{i}));
-    if isempty(sol)
-        if printLevel == 1 && ~parallelMode
-            fprintf('iteration %d.\n', i);
-        end
-        % do LP always
-        if allowLoops
-            LPsolution = solveCobraLP(LPproblem, cpxControl);
-        else
-            LPsolution = solveCobraMILP(addLoopLawConstraints(LPproblem, model));
-        end
-        % take the maximum flux from the flux vector, not from the obj -Ronan
-        % A solution is possible, so the only problem should be if its
-        % unbounded and if it is unbounded, the max flux is infinity.
-        if LPsolution.stat == 2
-            Flux = -LPProblem.osense * inf;
-        else
-            Flux = getObjectiveFlux(LPsolution, LPproblem);
-        end
+%get Number of reactions
+nRxns = numel(model.rxns);
+%Set the correct objective
+LPproblem.c = double(ismember(model.rxns,rxnNameList{i}));
+if isempty(sol)
+    if printLevel == 1 && ~parallelMode
+        fprintf('iteration %d.\n', i);
+    end
+    % do LP always
+    if allowLoops
+        LPsolution = solveCobraLP(LPproblem, cpxControl);
     else
-        LPsolution = sol;
-        Flux = getObjectiveFlux(LPsolution, LPproblem);        
+        LPsolution = solveCobraMILP(addLoopLawConstraints(LPproblem, model));
     end
-    % minimise the Euclidean norm of the optimal flux vector to remove loops -Ronan
-    if minNorm == 1
-        V = getMinNorm(LPproblem, LPsolution, nRxns, Flux, model, method);
+    % take the maximum flux from the flux vector, not from the obj -Ronan
+    % A solution is possible, so the only problem should be if its
+    % unbounded and if it is unbounded, the max flux is infinity.
+    if LPsolution.stat == 2

```

Comment:
You would have to check here, whether the flux is actually unbounded. 
If its not, then this is an error. Only if it is unbounded (in this direction), it is likely that this is a infinite solution.